modules/standard/Lists.chpl

```diff
@@ -0,0 +1,363 @@
+    */
+    proc extend(other: List(this._etype)) {}
+
+    /*
+      Extend this List by appending all the elements from an array.
+
+      :arg other: An array, the elements of which are appended to this List.
+    */
+    proc extend(other: this._etype[?d]) {}
+
+    /*
+      Insert an element at a given position in this List, shifting all elements
+      following that index one to the right. Note that `a.insert(a.size, x)`
+      is equivalent to `a.append(x)`.
+
+      :arg i: The index of the element at which to insert.
+      :arg x: The element to insert.
+
+      :throws IllegalArgumentError: If the given index is out of bounds.
+    */
+    proc insert(i: int, x: this._etype) throws {}
+
+    /*
+      Remove the first item from this List whose value is equal to _x_.
+
+      :arg x: The element to remove.
+
+      :throws IllegalArgumentError: If there is no such item.
+    */
+    proc remove(x: this._etype) throws {}
+
+    /*
+      Remove the item at the given position in this List, and return it. If no
+      index is specified, remove and return the last item in this List.
+
+      :arg i: The index of the element to remove. Defaults to the last item
+        in this List, or `a.remove(a.size - 1)`.
+      
+      :return: The item removed.
+
+      :throws IllegalArgumentError: If the given index is out of bounds.
+    */
+    proc pop(i: int=this._size - 1): this._etype throws {
+      return nil;
+    }
+
+    /*
+      Clear the contents of this List.
+    */
+    proc clear() {}
+
+    /*
+      Return a zero-based index into this List of the first item whose value
+      is equal to _x_.
+
+      :arg x: An element to search for.
+
+      :return: The index of the element to search for.
+
+      :throws IllegalArgumentError: If the given element cannot be found.
+    */
+    proc indexOf(x: _etype, start: int=0, end: int=_size): int throws {
+      return 0; 
+    }
+
+    /*
+      Return the number of times _x_ appears in this List.
+
+      :arg x: An element to count.
+
+      :return: The number of times a given element is found in this List.
+      :rtype: `int`
+    */
+    proc count(x: this._etype): int {
+      return 0;
+    }
+
+    /*
+      Sort the items of this List in place. The parameters of this method are
+      holdovers from Python. This method could more closely integrate with
+      Chapel sorting APIs.
+
+      :arg key: Unused in current API.
+      :arg reverse: True if this List should be sorted in reverse order.
+
+    */
+    proc sort(key: this._etype=nil, reverse: bool=false) {}
+
+    /*
+      Reverse the elements of this List in place.
+    */
+    proc reverse() {}
+
+    /*
+      Index this List via subscript. Returns a reference to the element at a
+      given index in this List.
+
+      :arg i: The index of the element to access.
+
+      :return: An element from this List.
+
+      :throws IllegalArgumentError: If the given index is out of bounds.
+    */
```

Comment:
Have we documented this/these/writeThis functions in other places?

Reply:
It seems like builtins would be documented in an obvious part of the Chapel Docs (I know we do here https://chapel-lang.org/docs/primers/specialMethods.html, but haven't checked the manual itself).